plugin/src/test/java/io/jenkins/plugins/casc/SchemaGenerationTest.java

```diff
@@ -0,0 +1,62 @@
+package io.jenkins.plugins.casc;
+
+import io.jenkins.plugins.casc.misc.JenkinsConfiguredWithCodeRule;
+import org.apache.log4j.spi.Configurator;
+import org.junit.Rule;
+import org.junit.Test;
+import java.util.*;
+
+
+import java.io.IOException;
+import java.util.LinkedHashSet;
+
+import static org.junit.Assert.*;
+
+public class SchemaGenerationTest {
+
+
+    @Rule
+    public JenkinsConfiguredWithCodeRule j = new JenkinsConfiguredWithCodeRule();
+
+
+    @Test
+    public void schemaShouldSucceed() {
+
+        LinkedHashSet linkedHashSet = new LinkedHashSet <> (ConfigurationAsCode.get().getRootConfigurators());
+        Iterator<RootElementConfigurator> i = linkedHashSet.iterator();
+
+        StringBuilder schemaString = new StringBuilder();
+        schemaString.append("{\n" +
+                "  \"$schema\": \"http://json-schema.org/draft-06/schema#\",\n" +
+                "  \"id\": \"http://jenkins.io/configuration-as-code#\",\n" +
+                "  \"description\": \"Jenkins Configuration as Code\",\n" +
+                "  \"type\": \"object\",\n" +
+                "  \"properties\": {\n" +
+                "");
+
+        while (i.hasNext()) {
+            List<Attribute> attributeList = new ArrayList<>(i.next().getAttributes());
+            for (Attribute a :attributeList) {
+                String type;
+                
+                if(a.type.toString().equals("int")){
+                     type = "int";
+                } else if (a.type.toString().equals("class java.lang.String")) {
+                    type = "string";
+                } else if (a.type.toString().equals("boolean")) {
+                    type = "boolean";
+                } else {
+                    type = "object";
+                }
+
+                schemaString.append(a.name + ": " + "{\n" +
+                        "      \"type\": \"" + type + "\",\n" +
+                        "      \"$ref\": \"#/definitions/" + a.type.getName() + "\"\n" +
+                        "    },");
+
+            }
+        }
+
+        System.out.println(schemaString);
```

Comment:
{
  "$schema": "http://json-schema.org/draft-06/schema#",
  "id": "http://jenkins.io/configuration-as-code#",
  "description": "Jenkins Configuration as Code",
  "type": "object",
  "properties": {
deprecated: {
      "type": "object",
      "$ref": "#/definitions/io.jenkins.plugins.casc.ConfigurationContext$Deprecation"
    },restricted: {
      "type": "object",
      "$ref": "#/definitions/io.jenkins.plugins.casc.ConfigurationContext$Restriction"
    },unknown: {
      "type": "object",
      "$ref": "#/definitions/io.jenkins.plugins.casc.ConfigurationContext$Unknown"
    },version: {
      "type": "object",
      "$ref": "#/definitions/io.jenkins.plugins.casc.ConfigurationContext$Version"
    },system: {
      "type": "object",
      "$ref": "#/definitions/com.cloudbees.plugins.credentials.SystemCredentialsProvider"
    },agentProtocols: {
      "type": "string",
      "$ref": "#/definitions/java.lang.String"
    },authorizationStrategy: {
      "type": "object",
      "$ref": "#/definitions/hudson.security.AuthorizationStrategy"
    },clouds: {
      "type": "object",
      "$ref": "#/definitions/hudson.slaves.Cloud"
    },crumbIssuer: {
      "type": "object",
      "$ref": "#/definitions/hudson.security.csrf.CrumbIssuer"
    },disableRememberMe: {
      "type": "boolean",
      "$ref": "#/definitions/boolean"
    },globalNodeProperties: {
      "type": "object",
      "$ref": "#/definitions/hudson.slaves.NodeProperty"
    },labelString: {
      "type": "string",
      "$ref": "#/definitions/java.lang.String"
    },markupFormatter: {
      "type": "object",
      "$ref": "#/definitions/hudson.markup.MarkupFormatter"
    },mode: {
      "type": "object",
      "$ref": "#/definitions/hudson.model.Node$Mode"
    },myViewsTabBar: {
      "type": "object",
      "$ref": "#/definitions/hudson.views.MyViewsTabBar"
    }
}

Reply:
yeah I think, it's likely our best option for now at least to start with.